src/unitxt/processors.py

```diff
@@ -277,3 +279,57 @@ def process_value(self, text: Any) -> Any:
 
         except:
             return 0
+
+
+class InferDictsToBinaryLogprobs(FieldOperator):
+    binary_class_names: list[str]
+
+    def process_value(self, obj: Any) -> Any:
+        try:
+            n_first_tokens = np.min([3, len(obj)])
+        except:
+            return []
+
+        for i in range(n_first_tokens):
+            try:
+                pos_probs, neg_probs = self.get_pos_neg_probs(pred_dict=obj[i])
+                if pos_probs or neg_probs:
+                    sum_probs = sum(pos_probs) + sum(neg_probs)
+                    if sum_probs > 0.0001:
+                        return sum(pos_probs) / sum_probs
+            except:
+                pass
+        return []
+
+    def get_pos_neg_probs(self, pred_dict):
+        token_logprobs = pred_dict["top_tokens"]
+        pos_probs = [
+            np.exp(d["logprob"])
+            for d in token_logprobs
+            if self.binary_class_names[1].lower() in d["text"].lower()
+        ]
+        neg_probs = [
+            np.exp(d["logprob"])
+            for d in token_logprobs
+            if self.binary_class_names[0].lower() in d["text"].lower()
+        ]
+        return pos_probs, neg_probs
+
+
+class InferDictsToBinaryLogProbsLastToken(InferDictsToBinaryLogprobs):
+    def process_value(self, obj: Any) -> Any:
+        try:
+            n_last_tokens = min([3, len(obj)])
+        except:
+            return []
+
+        for i in range(-1, -(n_last_tokens + 1), -1):
```

Comment:
rather than duplicate the code between the classes, you can have a  only method to return the range you traverse .

Reply:
done